addons/document/document.py

```diff
@@ -108,12 +108,27 @@ def search(self, cr, uid, args, offset=0, limit=None, order=None, context=None,
         for parent_id in visible_parent_ids:
             ids.extend(parents[parent_id])
 
+        set_ids = set(ids)
+
         # sort result according to the original sort ordering
         if count:
-            return len(ids)
+            result = len(set_ids)
         else:
-            set_ids = set(ids)
-            return [id for id in orig_ids if id in set_ids]
+            result = [id for id in orig_ids if id in set_ids]
```

Comment:
it might be worthwhile to compare the list comprehension to https://github.com/odoo/odoo/blob/09397384795f440b5924d9b325988e9cc6eb4d09/openerp/tools/misc.py#L1274